### PR TITLE
Dasharo/docs: Added documentation for Dasharo (UEFI) v0.1.0 for QEMU Q35

### PR DESCRIPTION
**Pull request by maheshtammisetti**
Added documentation for the Dasharo (UEFI) v0.1.0 for QEMU Q35.

### CODE REVIEW

File: blc [hunk 1]
```diff
@@ -0,0 +1 @@
+Error: Unhandled Rejection. Error: connect ECONNREFUSED 0.0.0.0:8000
```
Comment (miczyg1) [inline blc:1]: @maheshtammisetti I guess this is an unwanted file?

File: docs/variants/qemu_q35/building-manual.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#Dasharo (UEFI) v0.1.0 for QEMU Q35 - Building Manual
+
+In order to build the `Dasharo (UEFI) OVMF firmware image for QEMU Q35`, please follow the given steps to build the Dasharo firmware image from the EDK2 repository:
```
Comment (miczyg1) [inline docs/variants/qemu_q35/building-manual.md:3]: Please wrap all markdown documents (except markdown syntax) at 80 characters. Remember to do it every time from now on.

File: docs/variants/qemu_q35/building-manual.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    git clone https://github.com/Dasharo/edk2.git 
+    ```
+       
+2. Once it is done, please follow the below instructions, in order to prepare your environment for the building of the OVMF image.
+
+	
+	* Change the directory to Dasharo EDK2 repository locally - 
+	
+	```bash
+	cd Dasharo/edk2
```
Comment (miczyg1) [inline docs/variants/qemu_q35/building-manual.md:17]: THe path here will be just `edk2` after cloning. You don't mention anywhere that Dasharo directory is created.

File: docs/variants/qemu_q35/building-manual.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	```bash
+	source edksetup.sh
+	```
+								       
+	* Update the submodules in order get the latest dependencies and avoid any build process errors while building the OVMF image. 
+	
+	```bash
+	git submodules update --init
+	```
+
```
Comment (miczyg1) [inline docs/variants/qemu_q35/building-manual.md:33]: `make -C BaseTools` has to be invoked too, otherwise `build` command will not be recognized.

File: docs/variants/qemu_q35/building-manual.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	source edksetup.sh
+	```
+								       
+	* Update the submodules in order get the latest dependencies and avoid any build process errors while building the OVMF image. 
+	
+	```bash
+	git submodules update --init
+	```
+									       
+3. In order to build the UEFI firmware image it can be done in two ways, one way is to give the flags with `-D SMM_REQUIRE` (SMM was actually enabled while building the firmware image, but the flags differ with feature to feature, so please check the OvmfPkgX64.dsc file). And second way is to set the flags in `OvmfPkgX64.dsc` and then build the firmware image. For the purpose of this documentation, I have preferred to set the features in `OvmfPkgX64.dsc` directly.
```
Comment (miczyg1) [inline docs/variants/qemu_q35/building-manual.md:34]: I think, there is no reason to say the we have two methods. If we commit changes to DSC file, then let's skip the explanation.

File: docs/variants/qemu_q35/building-manual.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+In order to build the `Dasharo (UEFI) OVMF firmware image for QEMU Q35`, please follow the given steps to build the Dasharo firmware image from the EDK2 repository:
+
+1. Clone the official Dasharo EDK2 repository to your local instance with git or downloading the source code from github.
+
+    ```bash
+    git clone https://github.com/Dasharo/edk2.git 
+    ```
+       
+2. Once it is done, please follow the below instructions, in order to prepare your environment for the building of the OVMF image.
```
Comment (miczyg1) [inline docs/variants/qemu_q35/building-manual.md:11]: Do you foresee a docker environment? Currently, these procedures may or may not work, depending on what host operating system one will use to build EDK2.  EDK2 has some other dependencies that may not always be installed on other machines.

File: docs/variants/qemu_q35/development.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
+# Dasharo (UEFI) v0.1.0 for QEMU Q35
+
+
+1.  Please verify your qemu installation, and then invoke the qemu emulator with the built OVMF firmware image as given below:
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:4]: Would be good to give some references how to setup the QEMU installation.

File: docs/variants/qemu_q35/development.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+2. After executing the above qemu command, one should be able to see the UEFI built-in shell and should be able to get to the BIOS selection area.
+
+3. The features which are enabled in the `OvmfPkgX64.dsc` can be verified in the `BIOS menu` and also at the `Device Manager section` and the Dasharo features can be verified at the `Dasharo System Features` section.
+
+
+
+## Useful Tips for modifying the changes in .DSC & .FDF
+
+
+### Following the below guide, one can rebuild the firmware image and run it in the qemu and experiment with the features which are needed.
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:31]:
```suggestion
Following the below guide, one can rebuild the firmware image and run it in the
qemu and experiment with the features which are needed.
```

File: docs/variants/qemu_q35/development.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+
+
+1.  By making some changes in the `OvmfPkgX64.dsc` file, one can add features like OPAL, SATA security support for QEMU Q35 machine and also for specific platforms.
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:34]: Maybe these options should be permanently enabled... I mean we will always want to have OVMF fully 
featured, mainly for testing.

What I really meant by tips is how to quickly rebuild if you make changes in C files mostly.

File: docs/variants/qemu_q35/overview.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+* [Building manual](building-manual.md) - describes how to build Dasharo (UEFI) for
+    the QEMU Emulator Q35 Machine type.
+    
+* [Test matrix](test-matrix.md) - describes the validation scope used during the
+    Dasharo firmware validation procedure.
+    
+* [Development](development.md) - describes information for running the Dasharo OVMF in QEMU Q35 machine and for Dasharo Developers.
+
+
+
```
Comment (miczyg1) [inline docs/variants/qemu_q35/overview.md:15]:
```suggestion

```

Only 1 empty line of space between headers. Bulletpoint list should also not have empty spaces

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: blc (hunk 1 above)
Comment (maheshtammisetti) [reply blc:1]: Seems like an error generated, when I was trying to preview documentation webpage locally. Will remove it.

File: docs/variants/qemu_q35/development.md (hunk 8 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/development.md:31]: Noted. And will do it right away.

File: docs/variants/qemu_q35/development.md (hunk 9 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/development.md:34]: I agree, that the Dasharo Features should be enabled. But for someone who is building the Dasharo Ovmf, I thought it would be worth noting them to point where they can enable or disable the features.

File: docs/variants/qemu_q35/building-manual.md (hunk 5 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/building-manual.md:34]: Alright. Will do that.

File: docs/variants/qemu_q35/building-manual.md (hunk 4 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/building-manual.md:33]: Noted. Will work on it.

File: docs/variants/qemu_q35/building-manual.md (hunk 3 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/building-manual.md:17]: Will remove that.

File: docs/variants/qemu_q35/development.md (hunk 7 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/development.md:4]: Will do that, documentation to QEMU installation will do?

**Comment by pietrushnic** [comment]
@miczyg1 please do not review PRs that not pass pre-commit.

@maheshtammisetti 
* https://github.com/Dasharo/docs#pre-commit-hooks
* https://results.pre-commit.ci/run/github/316342881/1674488959.p5Dvf4udSPSV3ujjqr6tQw

Please use pre-commit.

File: docs/variants/qemu_q35/development.md [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+## Intro
+
+1. Please check your qemu setup and launch qemu outside
+docker instance (host machine) using built OVMF firmware image.
+
++ QEMU Installation:
+    - [Install QEMU on your linux distro](<https://www.qemu.org/download/#linux>)
+
+ ```bash
+ qemu-system-x86_64 -drive if=pflash,format=raw,file=Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd -debugcon file:debug.log -global isa-debugcon.iobase=0x402 -global ICH9-LPC.disable_s3=1 -net none -machine q35,smm=on
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:12]:
```suggestion
 qemu-system-x86_64 -drive if=pflash,format=raw,file=Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd \
      -debugcon file:debug.log -global isa-debugcon.iobase=0x402 -global ICH9-LPC.disable_s3=1 \
      -net none -machine q35,smm=on
```

Comment (miczyg1) [reply docs/variants/qemu_q35/development.md:12]: I believe there should be two FD files one for variables second for the code.

File: docs/variants/qemu_q35/development.md [hunk 12]
```diff
@@ -0,0 +1,138 @@
...
+ ```
+
++ `-drive` indicate device is pflash with firmware image of built OVMF.fd image.
+
++ `-debugcon file:debug.log -global isa-debugcon.iobase=0x402`
+default OVMF build writes debug messages to IO port **0x402**.
+The following qemu command line options save them in the file called debug.log.
+
++ `-global ICH9-LPC.disable_s3=1` SMM is put to use in the S3 suspend
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:21]: I think for S3 would be good to have it enabled.

File: docs/variants/qemu_q35/development.md [hunk 13]
```diff
@@ -0,0 +1,138 @@
...
++ `-net none` disables the OptionROM execution which
+will interfere with the firmware image, if not disabled.
+
++ `-machine q35,smm=on` For SMM to work,
+only Q35 machines are supported hence the machine type.
+
+1. After executing the above qemu command,
+   qemu boots into UEFI built-in shell and to BIOS selection area.
+
+1. The features which are enabled in `OvmfPkgX64.dsc` can be verified at `BIOS menu`.
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:34]: `BIOS Menu`? I don't recall anything like that in EDK2 setup application. If anything, I would call it `Setup menu`

File: docs/variants/qemu_q35/development.md [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+
+## Useful Tips for modifying the changes in .DSC & .FDF
+
+### Follow below steps, to rebuild the firmware image and experiments with features
+
++ If any code is added to C file or any library functions are invoked in other files,
+rebuild the OVMF image with the following command.
+
+```bash
+build -D DEBUG_ON_SERIAL_PORT -D BOOTLOADER=COREBOOT -a IA32 -a X64 -t GCC5 -b DEBUG -p OvmfPkg/OvmfPkgX64.dsc
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:46]:
```suggestion
build -D DEBUG_ON_SERIAL_PORT -a IA32 -a X64 -t GCC5 -b DEBUG -p OvmfPkg/OvmfPkgX64.dsc
```

We are not using coreboot here.

File: docs/variants/qemu_q35/development.md [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+```bash
+#
+# OPAL_PASSWORD
+#
+
+!if $(OPAL_PASSWORD_ENABLE) == TRUE
+TcgStorageCoreLib|SecurityPkg/Library/TcgStorageCoreLib/TcgStorageCoreLib.inf
+TcgStorageOpalLib|SecurityPkg/Library/TcgStorageOpalLib/TcgStorageOpalLib.inf
+!endif
+S3BootScriptLib|MdePkg/Library/BaseS3BootScriptLibNull/BaseS3BootScriptLibNull.inf
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:77]: Please align that with my review from edk2 PR. NULL S3BootScriptLib is redundant here I believe.

File: docs/variants/qemu_q35/development.md [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+!endif
+#
+#
+```
+
+## Dasharo System Features
+
+1. The .FDF file tells about location of source file and options used for build process.
+
+2. Below code snippet shows about the location of `SATA_PASSWORD` .inf file.
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:100]:
```suggestion
2. Below code snippet shows about the location of  SATA disk password `.inf` file.
```

File: docs/variants/qemu_q35/development.md [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+!if $(SATA_PASSWORD_ENABLE) == TRUE
+SecurityPkg/HddPassword/HddPasswordPei.inf
+!endif
+#
+#
+```
+
+## Dasharo System Features
+
+1. The .FDF file tells about location of source file and options used for build process.
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:98]:
```suggestion
1. The `.fdf` file tells about the location of the source file and options used for the build process.
```

You should use https://app.grammarly.com/ to check the grammar errors

File: docs/variants/qemu_q35/development.md [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+2. Below code snippet shows about the location of `SATA_PASSWORD` .inf file.
+
+```bash
+!if $(SATA_PASSWORD_ENABLE) == TRUE
+INF SecurityPkg/HddPassword/HddPasswordDxe.inf
+!endif
+```
+
++ Include all necessary source file locations
+to point at source libraries in .FDF file.
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:109]:
```suggestion
+ Include all necessary source file locations to point at source libraries in
`.fdf` file.
```

Please wrap exactly at 80 chars or the closets to 80 chars so that you fill most of the line. Repeat for every occurrence please

File: docs/variants/qemu_q35/test-matrix.md [hunk 19]
```diff
@@ -0,0 +1,48 @@
...
+| No.  | Supported test suite                              | Test suite ID | Supported test cases                 |
+|:-----|:--------------------------------------------------|:-------------:|:-------------------------------------|
+| 1.   | [UEFI compatible interface][EFI]                  | EFI           | All                                  |
+| 2.   | [UEFI Shell][USH]                                 | USH           | All                                  |
+| 3.   | [Display ports and LCD support][DSP]              | DSP           | DSP001.001      |
+| 4.   | [Custom boot logo][CLG]                           | CLG           | All                                  |
+| 5.  | [Debian Stable and Ubuntu LTS support][LBT]       | LBT           | LBT001.001, LBT001.002, LBT002.001, LBT002.002, LBT003.001, LBT003.002, LBT004.001, LBT004.002|
+| 6.  | [Windows booting][WBT]                            | WBT           | All                                  |
+| 7.  | [Custom Boot Keys][CBK]                           | CBK           | All                                  |
+| 8.  | [Qubes OS Support][QBS]                        | QBS           | QBS001.001, QBS001.002 |
```
Comment (miczyg1) [inline docs/variants/qemu_q35/test-matrix.md:23]: This may not be feasible to test under emulation. Please remove

File: fixme.txt [hunk 20]
```diff
@@ -1,81 +0,0 @@
-docs/variants/clevo_nv41/openness_analysis.md:5:81 MD013/line-length Line length [Expected: 80; Actual: 131]
```
Comment (miczyg1) [inline fixme.txt:1]: Don't remove this file. It seems like it was automatically generated and someone committed it to the repo. If anything, it should be fixed in a separate PR.

File: mkdocs.yml [hunk 21]
```diff
@@ -144,6 +144,12 @@ nav:
           - 'Recovery': variants/protectli_vp46xx/recovery.md
           - 'Hardware Configuration Matrix': variants/protectli_vp46xx/hardware-matrix.md
           - 'Test matrix': variants/protectli_vp46xx/test-matrix.md
+      - 'QEMU Q35 Emulator':
+          - 'Overview': variants/qemu_q35/overview.md
+          - 'Releases': variants/qemu_q35/releases.md
+          - 'Building manual': variants/qemu_q35/building-manual.md
+          - 'Development': variants/qemu_q35/development.md
+          - 'Test matrix': variants/qemu_q35/test-matrix.md
```
Comment (miczyg1) [inline mkdocs.yml:152]: overview_url: http://docs.dasharo.com/variants/protectli_vp46xx/overview/
        building_url: http://docs.dasharo.com/variants/protectli_vp46xx/building-manual/
        deploy_url: https://docs.dasharo.com/variants/protectli_vp46xx/initial-deployment/
        update_url: http://docs.dasharo.com/variants/protectli_vp46xx/firmware-update/
        recovery_url: http://docs.dasharo.com/variants/protectli_vp46xx/recovery/
        
We also need links to `Hardware Configuration Matrix`, `Initial deployment`, `Recovery` and `Firmware update` for our release automation to be satisfied. These section may contain just one sentence:

QEMU is an emulation environment and not a real hardware thus firmware {deployment, update, recovery} is not applicable.

Although we would probably need a Hardware Configuration Matrix to be filled with standard information about devices emulated by QEMU.

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: docs/variants/qemu_q35/development.md (hunk 11 above)
Comment (pietrushnic) [reply docs/variants/qemu_q35/development.md:12]: Of course otherwise variables are volatile. Please check OST2 Arch4021 2023_v1

File: docs/variants/qemu_q35/development.md (hunk 12 above)
Comment (pietrushnic) [reply docs/variants/qemu_q35/development.md:21]: I guess this is not that simple, I had problems booting with S3 enabled. Question is how S3 works in QEMU. In general this seem to be some workaround which I was not able to decode why it is needed. If we would figure out we should document that and probably propagate to OST2 Arch4021.

More to that at some point we have to align Dasharo with OST2, but this is other topic and longer term action. Maybe we can do that as part of Arch4221 UEFI Secure Boot

File: docs/variants/qemu_q35/building-manual.md [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+## Building the firmware image
+
+1. To build firmware image with required features set the flags with **TRUE/FLASE**.
+Following is an example of SMM feature enabled,
+`DEFINE SMM_REQUIRE = TRUE` in the **OvmfPkgX64.dsc** file.
+
+ Check the following build command:
+
+```bash
+build -D DEBUG_ON_SERIAL_PORT -D BOOTLOADER=COREBOOT -a IA32 -a X64 -t GCC5 -b DEBUG -p OvmfPkg/OvmfPkgX64.dsc.
```
Comment (miczyg1) [inline docs/variants/qemu_q35/building-manual.md:53]:
```suggestion
build -D DEBUG_ON_SERIAL_PORT -D BOOTLOADER=COREBOOT -a IA32 -a X64 -t GCC5 -b DEBUG -p OvmfPkg/OvmfPkgX64.dsc
```

File: docs/variants/qemu_q35/development.md (hunk 11 above)
Comment (miczyg1) [reply docs/variants/qemu_q35/development.md:12]:
```suggestion
 qemu-system-x86_64 -machine q35,smm=on \
-global driver=cfi.pflash01,property=secure,value=on \
-drive if=pflash,format=raw,unit=0,file=Build/OvmfX64/DEBUG_GCC5/FV/OVMF_CODE.fd,readonly=on \
-drive if=pflash,format=raw,unit=1,file=Build/OvmfX64/DEBUG_GCC5/FV/OVMF_VARS.fd \
-debugcon file:debug.log -global isa-debugcon.iobase=0x402 \
-global ICH9-LPC.disable_s3=
```

This should be the correct command. Also removed the `-net none` as we will want to use network boot as well.

File: docs/variants/qemu_q35/development.md [hunk 23]
```diff
@@ -0,0 +1,138 @@
+# Dasharo (UEFI) v0.1.0 for QEMU Q35
+
+## Intro
+
+1. Please check your qemu setup and launch qemu outside
+docker instance (host machine) using built OVMF firmware image.
+
++ QEMU Installation:
+    - [Install QEMU on your linux distro](<https://www.qemu.org/download/#linux>)
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:9]:
```suggestion
    - [Install QEMU on your linux distro](https://www.qemu.org/download/#linux)
```

File: docs/variants/qemu_q35/building-manual.md [hunk 24]
```diff
@@ -0,0 +1,66 @@
...
+- `DEBUG_ON_SERIAL_PORT` = The following flag must be given for the **DEBUG** build
+as the output is redirected to the I/O port **0x402** to qemu monitor.
+
+- Debug messages are visible on given I/O port.
+**0x402** will be later utilized for qemu invocation further in this documentation.
+
+- Once the build is completed, the OVMF firmware image can be found below given path:
+
+```bash
+edk2/Build/Ovmf/DEBUG_GCC5/Ovmf.fd
```
Comment (miczyg1) [inline docs/variants/qemu_q35/building-manual.md:65]:
```suggestion
edk2/Build/OvmfX64/DEBUG_GCC5/FV/OVMF_CODE.fd
edk2/Build/OvmfX64/DEBUG_GCC5/FV/OVMF_VARS.fd
```

File: docs/variants/qemu_q35/development.md [hunk 25]
```diff
@@ -0,0 +1,138 @@
...
+
+ ```bash
+ qemu-system-x86_64 -drive if=pflash,format=raw,file=Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd -debugcon file:debug.log -global isa-debugcon.iobase=0x402 -global ICH9-LPC.disable_s3=1 -net none -machine q35,smm=on
+ ```
+
++ `-drive` indicate device is pflash with firmware image of built OVMF.fd image.
+
++ `-debugcon file:debug.log -global isa-debugcon.iobase=0x402`
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply docs/variants/qemu_q35/development.md:17]: Ohh it is becasue of `DEBUG_ON_SERIAL_PORT` in the build command. We should remove it then.

File: docs/variants/qemu_q35/building-manual.md [hunk 26]
```diff
@@ -0,0 +1,66 @@
...
+`DEFINE SMM_REQUIRE = TRUE` in the **OvmfPkgX64.dsc** file.
+
+ Check the following build command:
+
+```bash
+build -D DEBUG_ON_SERIAL_PORT -D BOOTLOADER=COREBOOT -a IA32 -a X64 -t GCC5 -b DEBUG -p OvmfPkg/OvmfPkgX64.dsc.
+```
+
+- `DEBUG_ON_SERIAL_PORT` = The following flag must be given for the **DEBUG** build
+as the output is redirected to the I/O port **0x402** to qemu monitor.
```
Comment (miczyg1) [inline docs/variants/qemu_q35/building-manual.md:57]: It doesn;t have to be given. Moreover it shouldn't be given, because t breaks the debug output to the file. If we do not want any debug output we should build release image.

**Comment by miczyg1** [comment]
@maheshtammisetti I think I have pin pointed all issues. Please address all discussions so we can merge the PR.

File: docs/variants/qemu_q35/development.md (hunk 25 above)
[the remark this answers is not in the document]
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/development.md:17]: Yes, I had the DEBUG_ON_SERIAL_PORT enabled while I was building the OVMF image. I will remove it for the RELEASE build.

File: docs/variants/qemu_q35/building-manual.md (hunk 26 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/building-manual.md:57]: I thought it's helpful for people who want to debug or want to know how to debug just in case. I will make sure to remove it.

File: docs/variants/qemu_q35/test-matrix.md [hunk 27]
```diff
@@ -0,0 +1,48 @@
...
+
+| No.  | Supported test suite                              | Test suite ID | Supported test cases                 |
+|:-----|:--------------------------------------------------|:-------------:|:-------------------------------------|
+| 1.   | [Secure Boot support][SBO]                        | SBO           | All                                  |
+| 2.   | [ME neuter support][MNE]                          | MNE           | All                                  |
+| 3.   | [BIOS lock support][BLS]                          | BLS           | All                                  |
+
+[SBO]: ../../unified-test-documentation/dasharo-security/206-secure-boot.md
+[MNE]: ../../unified-test-documentation/dasharo-security/20F-me-neuter.md
+[BLS]: ../../unified-test-documentation/dasharo-security/20J-bios-lock-support.md
```
Comment (miczyg1) [inline docs/variants/qemu_q35/test-matrix.md:48]:
```suggestion

[SBO]: ../../unified-test-documentation/dasharo-security/206-secure-boot.md
```

File: docs/variants/qemu_q35/building-manual.md (hunk 26 above)
Comment (miczyg1) [reply docs/variants/qemu_q35/building-manual.md]: I meant:

`DEBUG_ON_SERIAL_PORT` can't be passed to the build command because it breaks the debug output to 0x402 port and the file itself.

So debug output to port 0x402 to a file works only if  `DEBUG_ON_SERIAL_PORT` is not passed. At least this is what I have observed

File: docs/variants/qemu_q35/development.md [hunk 28]
```diff
@@ -6,10 +6,10 @@
 docker instance (host machine) using built OVMF firmware image.
 
 + QEMU Installation:
-    - [Install QEMU on your linux distro](<https://www.qemu.org/download/#linux>)
+    - [Install QEMU on your linux distro](https://www.qemu.org/download/#linux)
 
  ```bash
- qemu-system-x86_64 -drive if=pflash,format=raw,file=Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd -debugcon file:debug.log -global isa-debugcon.iobase=0x402 -global ICH9-LPC.disable_s3=1 -net none -machine q35,smm=on
+ qemu-system-x86_64 -drive if=pflash,format=raw,file=Build/OvmfX64/RELEASE_GCC5/FV/OVMF.fd -debugcon file:debug.log -global ICH9-LPC.disable_s3=1 -net none -machine q35,smm=on
```
Comment (miczyg1) [inline docs/variants/qemu_q35/development.md:12]: For development it is desirable to keep using DEBUG build. Why have you changed it to RELEASE? I haven't asked for that...

File: docs/variants/qemu_q35/building-manual.md [hunk 29]
```diff
@@ -50,17 +50,12 @@ Following is an example of SMM feature enabled,
  Check the following build command:
 
 ```bash
-build -D DEBUG_ON_SERIAL_PORT -D BOOTLOADER=COREBOOT -a IA32 -a X64 -t GCC5 -b DEBUG -p OvmfPkg/OvmfPkgX64.dsc.
+build -a IA32 -a X64 -t GCC5 -b DEBUG -p OvmfPkg/OvmfPkgX64.dsc.
```
Comment (miczyg1) [inline docs/variants/qemu_q35/building-manual.md:53]:
```suggestion
build -a IA32 -a X64 -t GCC5 -b RELEASE -p OvmfPkg/OvmfPkgX64.dsc.
```

The build manual should use RELEASE version. The DEBUG version should be used in the development section.

File: docs/variants/qemu_q35/hardware-matrix.md [hunk 30]
```diff
@@ -0,0 +1,29 @@
+# Hardware configuration matrix
+
+## Introduction
+
+This document describes the devices emulated by the QEMU Q35.
+
+## QEMU Q35 //TBD
```
Comment (miczyg1) [inline docs/variants/qemu_q35/hardware-matrix.md:7]: When do you plan to add actual information about QEMU emulated hardware?

File: fixme.txt (hunk 20 above)
Comment (miczyg1) [reply fixme.txt:1]: @maheshtammisetti please bring back the file

File: docs/variants/qemu_q35/development.md (hunk 17 above)
Comment (miczyg1) [reply docs/variants/qemu_q35/development.md]: Now it is even worse... You didn't use grammarly as I have requested.
This is not gramatically correct sentence:
```
1. The `.fdf` describes about source file location and variables used during
build process.
```
That is how it should look like:

```
1. The `.fdf` describes the source files' location and variables used during the
build process.
```

File: docs/variants/qemu_q35/development.md (hunk 11 above)
Comment (miczyg1) [reply docs/variants/qemu_q35/development.md]: @maheshtammisetti you haven't applied it yet

File: docs/variants/qemu_q35/building-manual.md (hunk 6 above)
Comment (miczyg1) [reply docs/variants/qemu_q35/building-manual.md]: Still there is not command to start docker container, you have to add it too.

File: docs/variants/qemu_q35/building-manual.md [hunk 31]
```diff
@@ -0,0 +1,61 @@
...
+```bash
+git clone https://github.com/Dasharo/edk2.git
+```
+
+1. Follow below instructions, to prepare your environment for building OVMF image.
+
+- Setup the environment variables with the following command -
+
+```bash
+make -C BaseTools source edksetup.sh
```
Comment (miczyg1) [inline docs/variants/qemu_q35/building-manual.md:34]:
```suggestion
make -C BaseTools
source edksetup.sh
```

**Comment by miczyg1** [comment]
@maheshtammisetti please reply in the converstations whether you have applied my suggestions. I am wasting a lot of time searching for the placed to check whether you have changed them or not...

**Comment by pietrushnic** [comment]
@artur-rs In https://docs.dasharo.com/newcomers/ we should clearly explain that expected reply to review conversation (when it is possible, there some exceptions) should be:

```
Fixed: <link to commit?
```

File: docs/variants/qemu_q35/building-manual.md (hunk 2 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/building-manual.md]: This is fixed,

File: docs/variants/qemu_q35/hardware-matrix.md (hunk 30 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/hardware-matrix.md:7]: The problem is with the internet connectivity to the Ubuntu with OVMF, I am working on it (by trying libvirt as of now), due to the known issue of optionROM.

Comment (pietrushnic) [reply docs/variants/qemu_q35/hardware-matrix.md:7]: @maheshtammisetti OST2 Arch4021 should teach you all thing required for networking configuration in QEMU. `The problem` is not correct explanation of the error you face please be more precise when you describing problems.

Comment (miczyg1) [reply docs/variants/qemu_q35/hardware-matrix.md:7]: > due to the known issue of optionROM

The issue is not know unless you add it to https://github.com/Dasharo/dasharo-issues/issues with the explanation what is happening, error logs, etc. Now you are trying some libvirt method which we have no idea if it will work and leaving the original problem unsolved...

File: docs/variants/qemu_q35/building-manual.md (hunk 6 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/building-manual.md]: Added command to start docker container, followed by where to the build firmware image using QEMU

File: docs/variants/qemu_q35/development.md (hunk 11 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/development.md]: Applied the fix

File: docs/variants/qemu_q35/development.md (hunk 17 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/development.md]: Used grammarly to fix the sentence.

File: fixme.txt (hunk 20 above)
Comment (maheshtammisetti) [reply fixme.txt:1]: Added back the fixme.txt file please check it.

File: docs/variants/qemu_q35/building-manual.md (hunk 26 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/building-manual.md]: Yes without the DEBUG_ON_SERIAL_PORT command the debug log is generated and can view the logs.

File: docs/variants/qemu_q35/development.md (hunk 28 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/development.md:12]: I reverted the changes and fixed them accordingly.

File: docs/variants/qemu_q35/building-manual.md (hunk 29 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/building-manual.md:53]: Fixed according to the review.

File: docs/variants/qemu_q35/development.md [hunk 32]
```diff
@@ -0,0 +1,138 @@
...
+1. After executing the above qemu command,
+   qemu boots into UEFI built-in shell and to BIOS selection area.
+
+1. The features which are enabled in `OvmfPkgX64.dsc` can be verified at `Setup menu`.
+   At `Device Manager section` Dasharo features can
+   be verified in `Dasharo System Features` section.
+
+## Useful Tips for modifying the changes in .DSC & .FDF
+
+### Follow below steps, to rebuild the firmware image and experiments with features
```
[the remark this answers is not in the document]
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/development.md:40]: I have applied the changes.

File: docs/variants/qemu_q35/building-manual.md (hunk 31 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/building-manual.md:34]: I have applied the changes.

File: docs/variants/qemu_q35/hardware-matrix.md (hunk 30 above)
Comment (maheshtammisetti) [reply docs/variants/qemu_q35/hardware-matrix.md]: Added Hardware matrix after the fix of OptionROM issue

**Comment by artur-rs** [comment]
> @artur-rs In https://docs.dasharo.com/newcomers/ we should clearly explain that expected reply to review conversation (when it is possible, there some exceptions) should be"

@pietrushnic addressed here: https://github.com/Dasharo/docs/pull/548

**Comment by pietrushnic** [comment]
@artur-rs thanks. @maheshtammisetti please follow https://docs.dasharo.com/newcomers/#dasharo-contribution

File: docs/variants/qemu_q35/building-manual.md [hunk 33]
```diff
@@ -0,0 +1,69 @@
...
+
+```bash
+build -a IA32 -a X64 -t GCC5 -b RELEASE -p OvmfPkg/OvmfPkgX64.dsc
+```
+
+- Once the build is completed, the OVMF firmware image can be found below given path:
+
+```bash
+edk2/Build/Ovmf/DEBUG_GCC5/FV/OVMF_CODE.fd
+edk2/Build/Ovmf/DEBUG_GCC5/FV/OVMF_VARS.fd
```
Comment (miczyg1) [inline docs/variants/qemu_q35/building-manual.md:68]:
```suggestion
edk2/Build/Ovmf/RELEASE_GCC5/FV/OVMF_CODE.fd
edk2/Build/Ovmf/RELEASE_GCC5/FV/OVMF_VARS.fd
```

**Review by miczyg1 — APPROVED** [review]

**Comment by miczyg1** [comment]
I have cleaned up and fixed the reamining comments
